src/JWT.php

```
@@ -199,10 +199,16 @@ public static function decode(
     public static function encode(
         array $payload,
         $key,
-        string $alg,
+        ?string $alg = null,
         string $keyId = null,
         array $head = null
     ): string {
+        if (is_a($key, Key::class)) {
+            $alg = $key->getAlgorithm();
+            $key = $key->getKeyMaterial();
```

We would also want to throw an exception if `Key` isn't `null`.  Otherwise this would be possible and potentially cause confusion:

```php
JWT::encode($payload, $key, 'FooAlg');
```